app/components/category-item.js

```diff
@@ -15,4 +18,37 @@ export default Ember.Component.extend({
     let userCategories = this.get('userCategories');
     return userCategories.hasCategory(category);
   }),
+
+  actions: {
+    addCategory(category) {
+      this.set('isLoading', true);
+      this.addCategory(category).catch(() => {
```

Comment:
Ah, got it. I'm not sure I absolutely love that, though. Having it one place is great, but It wasn't clear to me we're passing it down, for example.

Is it possible to change the name while passing it down, to make it clearer where it originates from? Alternatively, maybe `addCategory` and `removeCategory` could be functions of the `currentUser` service?